dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2130,161 @@ def _update_chi2(self):
+            sample_size = getattr(compiler, "sample_size")
+            true_count = sample_size - null_count
+
+            # null_count and sample_size get updated on profile_update
+            # Therefore the priors are calculated from the updated values
+            prior_null = null_count / sample_size
+            prior_not_null = true_count / sample_size
+
+            # Partition data based on whether target column value is null or not
+            col_is_null = data.iloc[:, col_id].isnull()
+            partition_null = data[col_is_null]
+            partition_not_null = data[~col_is_null]
+
+            # Drop the target column from partitions
+            partition_null = partition_null.drop(columns=col_id)
+            partition_not_null = partition_not_null.drop(columns=col_id)
+
+            # Calculate mean vectors of each partition
+            # i.e mean vectors of rows where target col is or is not null
+            mean_not_null, mean_null = (
+                partition_not_null.mean().tolist(),
+                partition_null.mean().tolist(),
+            )
+
+            # Combine with old means
+            if col_id in self._null_replication_metrics:
+                prev_null_count, prev_sample_size = prev_counts[col_id]
+                prev_true_count = prev_sample_size - prev_null_count
+
+                # null_count, sample_size and true_count of new data chunk
+                added_null_count, added_sample_size = (
+                    null_count - prev_null_count,
+                    sample_size - prev_sample_size,
+                )
+                added_true_count = added_sample_size - added_null_count
+
+                prev_class_means = self._null_replication_metrics[col_id]["class_mean"]
+                prev_mean_not_null, prev_mean_null = prev_class_means
+
+                mean_not_null = utils.combine_means(
+                    prev_mean_not_null, mean_not_null, prev_true_count, added_true_count
+                )
+                mean_null = utils.combine_means(
+                    prev_mean_null, mean_null, prev_null_count, added_null_count
+                )
+
+            self._null_replication_metrics[col_id] = {
+                # Array index serves as class label
+                # 0 indicates not null, 1 indicates null
+                "class_prior": [prior_not_null, prior_null],
+                "class_mean": [mean_not_null, mean_null],
+            }
+
+    def _merge_null_replication_metrics(self, other):
+        """
+        Merge null replication metrics between two data profiles
+
+        :param other: profile being added to this one.
+        :type other: StructuredProfiler
+        :return: merged nan replication metrics
+        :rtype: dict
+        """
+        merged_properties = {}
+        for col_id in range(len(self._profile)):
+            self_compiler = self._profile[col_id]
+            other_compiler = other._profile[col_id]
```

Comment:
I may have done `self_profile` and `other_profile` but won't hold the PR up over this